controller/github_user.go

```diff
@@ -0,0 +1,49 @@
+package controller
+
+import (
+	"fmt"
+	"lolidelgado/github-users/usecase"
+	"net/http"
+	"strconv"
+
+	"github.com/gorilla/mux"
+	"github.com/unrolled/render"
+)
+
+type GithubUser struct {
+	render *render.Render
+	g      *usecase.GithubUserUseCase
+}
+
+func NewGithubUser(r *render.Render, g *usecase.GithubUserUseCase) *GithubUser {
+	return &GithubUser{r, g}
+}
+
+func (c *GithubUser) GetGithubUsers(rw http.ResponseWriter, req *http.Request) {
+	switch req.Method {
+	case http.MethodGet:
+		githubUsers, err := c.g.FetchAll()
+		if err != nil {
+			c.render.Text(rw, http.StatusInternalServerError, fmt.Sprintf("This is not fine🔥\nAnd the reason is: %s", err))
+			return
+		}
+		c.render.JSON(rw, http.StatusOK, githubUsers)
+	}
+
+}
```

Comment:
As the comment in the `router.go file`, you can now get rid of this switch, as you already know this is a method specific for get request and it's already filtered at router level.

```suggestion
// GetGithubUser fetches all github users from the usecase
func (c *GithubUser) GetGithubUsers(rw http.ResponseWriter, req *http.Request) {
	githubUsers, err := c.g.FetchAll()
	if err != nil {
		c.render.Text(rw, http.StatusInternalServerError, fmt.Sprintf("This is not fine🔥\nAnd the reason is: %s", err))
		return
	}
	
	c.render.JSON(rw, http.StatusOK, githubUsers)
}
```